crates/bevy_ecs/src/query/state.rs

```diff
@@ -138,6 +138,34 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         state
     }
 
+    /// Creates a new [`QueryState`] from a given [`QueryBuilder`] and inherits it's [`FilteredAccess`].
+    pub fn from_builder(builder: &mut QueryBuilder<Q, F>) -> Self {
```

Comment:
Maybe actually impl `From<QueryBuilder> for QueryState`? That way APIs that want to be generic can accept `query: impl Into<QueryState>`.